osu.Game/Scoring/ScoreManager.cs

```diff
@@ -72,8 +76,31 @@ protected override ScoreInfo CreateModel(ArchiveReader archive)
             }
         }
 
+        private readonly Dictionary<string, User> previouslyLookedUpUsernames = new Dictionary<string, User>();
+
         protected override Task Populate(ScoreInfo model, ArchiveReader archive, CancellationToken cancellationToken = default)
-            => Task.CompletedTask;
+        {
+            // These scores only provide the user's username but we need the user's ID too.
+            if (model.UserID <= 1 && model.UserString != null)
```

Comment:
there's zero chance this is working. you're doing an async request and not blocking. work by exxample there's likely a simialr thing in beatmap import or oterhwise.